File: 0339A.Helpful.Maths/solution.py
def solution():
    line = input()

    counts = [0, 0, 0]
    for c in line:
        if c != "+":
            counts[int(c) - 1] += 1

    is_first = True
    for i in range(3):
        for count in range(counts[i]):
            if is_first:
                is_first = False
            else:
                print("+", end="")
            print(i + 1, end="")
    print("")


if __name__ == "__main__":
    solution()

File: 0339A.Helpful.Maths/solution.cpp
#include <iostream>
#include <string>


void solution() {
    std::string line;
    std::getline(std::cin, line);

    int counts[3] = {0, 0, 0};
    for (const char c: line) {
        if (c != '+') {
            counts[int(c - '0') - 1] += 1;
        }
    }

    bool is_first = true;
    for (int i = 0; i < 3; ++i) {
        for (int count = 0; count < counts[i]; ++count) {
            if (is_first) {
                is_first = false;
            } else {
                std::cout << '+';
            }

            std::cout << i + 1;
        }
    }

    std::cout << '\n';
}




void setup() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
}


int main() {
    setup();
    solution();
}
